TSQLLint.Lib/Rules/PredicateIsAlwaysTrue.cs

```diff
@@ -20,32 +20,26 @@ public PredicateIsAlwaysTrue(Action<string, string, int, int> errorCallback)
         
         public override void Visit(BooleanComparisonExpression node)
         {
-            {
-                var first = node.FirstExpression as Literal;
-                var second = node.SecondExpression as Literal;
+            var first = node.FirstExpression as Literal;
+            var second = node.SecondExpression as Literal;
 
-                if (first != null && second != null)
-                {
-                    if (string.Equals(first.Value, second.Value))
-                    {
-                        ErrorCallback(RULE_NAME, RULE_TEXT, node.StartLine, node.StartColumn);
-                    }
-                }
-            }
-            
+            if (first != null && second != null && string.Equals(first.Value, second.Value))
+	    {
+	        ErrorCallback(RULE_NAME, RULE_TEXT, node.StartLine, node.StartColumn);
+                return;
+	    }
+
+            var first = node.FirstExpression as ColumnReferenceExpression;

```

Comment:
Change still doesn't address the fact that if first != null and second != null you don't want to continue to then try to see if they are ColumnReferenceExpression.

Honestly, I'd break this into two smaller private methods. One for Literal and one for Column Reference Expression both called from Visit. I'd make the return value a boolean indicating whether or not it was able to be translated to the corresponding value. Then you could just use short circuit logic to not call the second method if it is the first one. If a third shows up that pattern would still hold.